### DISCUSSION TITLE
Thank you Gerry!

### DISCUSSION THREAD

**Discussion by mdehrlich** [GerryDazoo/Slinger#541] [General]
V4 with Slingerplayer is awesome.  Latency is much less than VLC or SMplayer.  I like playing video on laptop with remote on cellphone (cuts latency in half). The configs auto-populated in SP.  Very nice.  Suggestion, allow locking aspect ratio when resizing window.  Excitingly awaiting firestick version!

**Comment by GerryDazoo** [comment]
Thanks for the kind words. However I don't see how using the remote on your cell phone would be perceptually any faster than the integrated remote. Sending a HTTP Post message over the network takes just a few milliseconds. The lion's share of the latency is in the video buffering.

**Reply by GerryDazoo** [reply]
[SlingerPlayerFirestick.zip](https://github.com/GerryDazoo/Slinger/files/10794537/SlingerPlayerFirestick.zip)
Here's a start, not pretty but... I'm still missing the code to allow you to add your own URLs but it sounds like it might be usable for you.
Mini Manual: Left/Right buttons to select item. Up/Down on URL item to select Slingbox. The + looking control is FullScreen. A/R is aspect ratio.

**Reply by flipper2222222** [reply]
hi Gerry,
AFASIK some apps that can be loaded on Firesticks can also be loaded on a Google TV dongle, would your slingplayerfirestick be one of these apps?
thanks

**Reply by DVoorhees1957** [reply]
Hi Gerry, I installed the SlingerPlayer apk on my Fire TV, but for the life of me can't figure out how to get the Slinger URL entered/specified. There doesn't appear to be a way to bring up a keyboard, and clicking on the URL field (which just shows "http://" and nothing else) just yields the following message: "Slinger Player Error connecting to Slinger. Waiting to Play"

Screen shot follows:
![Slinger Player screen shot](https://user-images.githubusercontent.com/13420879/220467178-2c4cbbf5-7692-4787-bc13-f6b2e430d540.jpg)

**Reply by mdehrlich** [reply]
I used a stopwatch. I get 3 second delay (Florida to NJ).  Cellphone seems same as SP based remote.

**Reply by GerryDazoo** [reply]
Dvoor:
There's missing code. Please consider the post Pre-Beta.  For now, you need to be running the 4.0 server code to get it to auto-populate the list.

**Reply by DVoorhees1957** [reply]
Hmmm, I was running the 4.0 server code (downloaded 8 days ago, on 2/13/23 - has it changed since then?)... there was no auto-population...?

**Reply by GerryDazoo** [reply]
SP sends out a broadcast message on the LAN to find the server. You should see something like this in the server logs when it receives the broadcast.
Received GETSERVERCONFIG Request b'GETSERVERCONFIG' ('192.168.117.100', 58061) where 192.168.117.100 is the IP address of the Firestick.
I suspect your firewall is blocking the packet. try opening UDP port 50005.

**Reply by blasco403** [reply]
I installed and tried SP for Firestick, and I get the same thing, it does not populate the list from Slinger V4. I am remote from the Slinger Server but have a point to point VPN joining my two network subnets. I suspected it was a broadcast issue as my two networks are joined at layer 3 not 2.

**Reply by GerryDazoo** [reply]
Yes, quite possibly. Many VPN don't play nice with UDP.

**Reply by DVoorhees1957** [reply]
Gerry, I do see a GETSERVERCONFIG request, but it's pretty quickly followed by an error message: "ERROR While handling GETPORTNUMBERS request [WinError 10054] An existing connection was forcibly closed by the remote host". The full server log follows (I first tried an SMPlayer session, successfully, from the laptop hosting the Slingbox Server; the Firestick attempt is near the bottom of this log):
```
C:\Tools\Slinger\0 - v4.0>slingbox_server.py config.ini
Version : 4.0 Running on Windows-7-6.1.7601-SP1 pid= 10624 C:\Tools\Slinger\0 -
v4.0\slingbox_server.py
Using config file config.ini
Connection Manager Running on port 8080 with 2 max streams using URL slingbox.
BOXES [('sb500', 'sb500')]
BOX URL /slingbox/sb500
Building page for sb500
Using built in default remote page definition.
Streamer Running:  2 config.ini sb500 sb500 8080 8388608
starting up on  port 8080 8388608
sb500 Checking for slingbox at ('107.214.221.167', 55206) 0
sb500 ('107.214.221.167', 55206) OK
sb500 Using slingbox at  ('107.214.221.167', 55206)
Streamer:  sb500 Waiting for first stream, flushing any IR requests that arrive
while not connected to slingbox
 * Serving Flask app 'slingbox_server'
 * Debug mode: off
Broadcast Responder Running
Server IP address is 192.168.0.67
02/22/2023, 10:51:14.553   Streaming request from ('127.0.0.1', 51769)  /slingbo
x/sb500
STREAM=127.0.0.1:51769:0

Slinginfo  "350/500/M1/M2" 12 30 107.214.221.167 55206 3072 2 8AFvGWo2VvtpkLt
sb500 Starting Stream for  127.0.0.1:51769
sb500 Opening Control stream 0x2000 0 0
Connecting... ('107.214.221.167', 55206) Control
sb500 Selecting Video Source 2
sb500 Keycodes= [1, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 22
, 24, 25, 26, 27, 28, 29, 30, 31, 32, 34, 35, 37, 38, 39, 40, 41, 42, 43, 44, 46
, 47, 51]
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 1000 VideoSmoothne
ss= 0 IframeRate= 5 AudioBitRate= 64
Connecting... ('107.214.221.167', 55206) Stream
sb500 Stream started at 02/22/2023, 10:51:18.389  744 0
.........02/22/2023,10:52:52 sb500 1 Clients.127.0.0.1:51769
.........02/22/2023,10:54:24 sb500 1 Clients.127.0.0.1:51769
.........02/22/2023,10:55:59 sb500 1 Clients.127.0.0.1:51769
.........02/22/2023,10:57:35 sb500 1 Clients.127.0.0.1:51769
.........02/22/2023,10:59:10 sb500 1 Clients.127.0.0.1:51769
.........02/22/2023,11:00:46 sb500 1 Clients.127.0.0.1:51769
.........02/22/2023,11:02:18 sb500 1 Clients.127.0.0.1:51769
.........02/22/2023,11:03:52 sb500 1 Clients.127.0.0.1:51769
.........02/22/2023,11:05:26 sb500 1 Clients.127.0.0.1:51769
.........02/22/2023,11:07:02 sb500 1 Clients.127.0.0.1:51769
.........02/22/2023,11:08:37 sb500 1 Clients.127.0.0.1:51769
..02/22/2023, 11:09:00.984  sb500 Stream Terminated for  127.0.0.1:51769
sb500 Shutting down connections
sb500 Logging Out
Streamer:  sb500 Waiting for first stream, flushing any IR requests that arrive
while not connected to slingbox
Received GETSERVERCONFIG Request b'GETSERVERCONFIG' ('192.168.0.127', 57461)
ServerConfig port=8080,maxremotestreams=2,enableremote=yes,urlbase=slingbox|
ERROR While handling GETPORTNUMBERS request [WinError 10054] An existing connect
ion was forcibly closed by the remote host
```
The config.ini file follows, in case you need it:
```
[SLINGBOXES]
sb500=sb500

[sb500]
sbtype="350/500/M1/M2"
password=<removed>
ipaddress=107.214.221.167
port=55206
name=sb500
Resolution=12
VideoBandwidth=1000
FrameRate=30
VideoSmoothness=0
VideoSource=2
maxstreams=2

[SERVER]
; local port number for the server to listen on for connections
; Send all streaming and Remote request to this port number
port=8080
maxremotestreams=2
enableremote=yes
```

**Reply by GerryDazoo** [reply]
I think this could still be a firewall issue. Please open up UDP port 50005.
Do you know how to use Wireshark? A packet capture might be useful if changing the firewall settings don't help.

**Comment by mdehrlich** [comment]
Will firestick work over wan?

**Reply by GerryDazoo** [reply]
Yes, once I write the code to allow you to enter you're own URL. I'm working on it. What I posted was a pre-pre-release version.

**Reply by mdehrlich** [reply]
great :-)

**Reply by GerryDazoo** [reply]
Grab the apk I posted below. Use the Menu button (three bars) to get to the page where you can add your own URLs.
Use the arrow buttons to navigate. Then select to edit the entry.

**Reply by blasco403** [reply]
Not sure if I am missing something, but I only see the original zip file, not a new one you are referencing.

Thanks.

**Reply by GerryDazoo** [reply]
nope. I forgot to press reply on the other message. Should be there now

**Reply by blasco403** [reply]
Hello,

I have installed on two gen3 firesticks. Working well so far. Entering URL was easy, usual pop up keyboard.

Usage for others:

When in app, use round ring for navigation in menus. Left right to pick icons and move to URL field, up and down to select URL if you configure more than one. OK to select. 

To enter the URL's, navigate to URL field and press the menu button "3 dashes". Use navigation ring to navigate and OK to select. The keyboard will pop up as usual. It appears you can only enter 2 URL's. Once entered, select next and then press "return", it will prompt you to save changes.

URL'(s) are remembered between app launches.

Good luck.

**Reply by GerryDazoo** [reply]
You can only add one at a time. But you can do that multiple times. I had a 10 configured but most of them were auto-populated from the multiple servers on my LAN. There is no limit built into the program.

**Comment by RocketAlf** [comment]
@Gerry: I just tried to install SlingerPlayer on my FireTV Stick (2nd Gen) SW version "FireOS 5.2.9.4" using the app "Apps2Fire". But did not work :-(. The installation was aborted with the error code "INSTALL_FAILED_OLDER_SDK".

On Google I found the following explanation: "SDK doesn't really mean SDK, it means the platform only supports an older SDK than the app is using.  You need to make sure the app you are installing isn't built for a newer version of the platform."

That means I can NOT install SlingerPlayer on my older FireTV Stick (2nd Gen). Correct?

**Reply by GerryDazoo** [reply]
Probably for now. I have no way to test it here. You did do a check for updates?

**Reply by GerryDazoo** [reply]
GIve this a try. No guarantees....
[SlingerPlayer.zip](https://github.com/GerryDazoo/Slinger/files/10817405/SlingerPlayer.zip)

**Reply by RocketAlf** [reply]
Hi Gerry, thanks for the update. But is only a partial success :-(. The app can now be installed on my FireTV Stick, but it does not start. When I click on the icon, I see the SlingerPlayer logo for 1..2 seconds and then a black screen for 1 second. And then the app is "terminated" and I see the FireTV startup screen again. 

Something doesn't seem to fit yet ;-) to my "ancient" FireTV Stick (2nd Gen).

I had updated the software of the FireTV Stick before the installation.

**Reply by mdehrlich** [reply]
SP Running on my firestick.  Jittery at full HD.  OK on lower res.  A bit tricky to get into editing the URL.  Glad to have it.
Thanks again.

**Reply by GerryDazoo** [reply]
Not sure what you mean by tricky?
Menu to call up the URL page. Up and down arrows to select the URL then press the big select button to edit.
I had to bump down my bandwidth setting to 6000.

**Reply by mdehrlich** [reply]
thanks for the guidance.   Which bandwidth are we changing?  In the configuration?  Didn't think that did anything.

**Comment by floormanager1** [comment]
Using the Slingplayer app on Firestick and on Windows.  It's great well done Gerry.  Much improved over VLC.  Only issue is that my oldest Firestick will not install, get the same issues at RocketAlf.  Might be time for a new stick.
